src/SignalR/clients/java/signalr/src/main/java/com/microsoft/signalr/LongPollingTransport.java

```diff
@@ -75,18 +73,27 @@ public Completable start(String url) {
                     this.active = true;
                 }
                 this.threadPool = Executors.newCachedThreadPool();
-                threadPool.execute(() -> poll(url).subscribeWith(receiveLoop));
+                threadPool.execute(() -> {
+                    this.onReceiveThread = Executors.newSingleThreadExecutor();
```

Comment:
Why move this?

Reply:
Because if there is an error before the second poll, we don't need to make a thread 🤷‍♀️